cypher-shell/src/main/java/org/neo4j/shell/cli/CommandReader.java

```diff
@@ -16,20 +16,13 @@
 import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 import static java.lang.System.getProperty;
 
 public class CommandReader implements Historian {
```

Comment:
Having too many constructors seem a bit confusing. We can maybe inline the ones that are used only by the test.

The method `setupHistoryFile` can be pushed as init method elsewhere. It feels like CommandReader is doing too much as part of initiation of History. I'm not sure why CommandReader needs to manage History ex: setup, flush etc..